keras_preprocessing/image/utils.py

```diff
@@ -128,9 +128,17 @@ def load_img(path, grayscale=False, color_mode='rgb', target_size=None,
 
 
 def list_pictures(directory, ext='jpg|jpeg|bmp|png|ppm'):
+    """Lists all pictures in a directory, including all subdirectories.
+
+    # Arguments
+        directory: Absolute path to the directory
+        ext: extensions of the pictures, separated by '|'
+    # Returns
+        a list of paths
+    """
     return [os.path.join(root, f)
             for root, _, files in os.walk(directory) for f in files
-            if re.match(r'([\w\d-]+\.(?:' + ext + '))', f.lower())]
+            if re.match(r'.*(' + ext + ')$', f.lower())]
```

Comment:
I think it is clearer to do `f.endswith(tuple_of_extensions_with_dot)`, what do you think? perhaps is also faster

Reply:
Don't know if its faster, but clearer yeah. Should I also change the function parameter to be a tuple or should I split the string.

Reply:
Tuple is fine because it's immutable. Without the dot. Let's add the dot on the endswith logic. Please also check that it can receive a single extension from the user.

Reply:
seems to be faster

![selection_047](https://user-images.githubusercontent.com/16643700/52729797-27d1e380-2fba-11e9-9335-46bfab3e4dee.png)